assets/js/googlesitekit/data/create-notifications-store.js

```diff
@@ -108,13 +112,37 @@ export const createNotificationsStore = ( type, identifier, datapoint, {
 		 * @since 1.6.0
 		 * @private
 		 *
-		 * @return {Object} Redux-style action.
+		 * @return {Object} {response, error}
 		 */
-		fetchNotifications() {
-			return {
+		*fetchNotifications() {
+			let response, error;
+
+			yield {
 				payload: {},
-				type: FETCH_NOTIFICATIONS,
+				type: START_FETCH_NOTIFICATIONS,
 			};
+
+			try {
+				response = yield {
+					payload: {},
+					type: FETCH_NOTIFICATIONS,
+				};
+
+				yield actions.receiveNotifications( response );
+
+				yield {
+					payload: {},
+					type: FINISH_FETCH_NOTIFICATIONS,
+				};
+			} catch ( e ) {
+				error = e;
+				yield {
+					payload: {},
+					type: CATCH_FETCH_NOTIFICATIONS,
+				};
+			}
+
+			return { response, error };
```

Comment:
Yeah, it's definitely more cohesive 😄 

> One question, what is the purpose of returning the response and error here?

It's necessary for use in the resolver or other actions where further logic is dependent on success/fail or those values specifically. Currently, state only has a single `error` key, so if we were to use that instead of what is returned, it could lead to bugs where any error could be interpreted as the error from a request/action.

Also, there are cases where we may want to add some data to the store in a resolver, rather than every time its respective fetch happens. I think `matchedProperty` handling may be one such case.